### PR TITLE
Add Support for OpenSearch Vector database

### PR DESCRIPTION
**Pull request by naveentatikonda**
### Description
This PR adds a wrapper which adds support for the OpenSearch vector database. Using opensearch-py client we are ingesting the embeddings of given text into opensearch cluster using Bulk API. We can perform the `similarity_search` on the index using the 3 popular searching methods of OpenSearch k-NN plugin:

- `Approximate k-NN Search` use approximate nearest neighbor (ANN) algorithms from the [nmslib](https://github.com/nmslib/nmslib), [faiss](https://github.com/facebookresearch/faiss), and [Lucene](https://lucene.apache.org/) libraries to power k-NN search.
- `Script Scoring` extends OpenSearch’s script scoring functionality to execute a brute force, exact k-NN search.
- `Painless Scripting` adds the distance functions as painless extensions that can be used in more complex combinations. Also, supports brute force, exact k-NN search like Script Scoring.

### Issues Resolved 
https://github.com/hwchase17/langchain/issues/1054

### CODE REVIEW

File: docs/modules/indexes/chain_examples/vector_db_qa_with_sources.ipynb [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "from langchain.embeddings.cohere import CohereEmbeddings\n",
     "from langchain.text_splitter import CharacterTextSplitter\n",
     "from langchain.vectorstores.elastic_vector_search import ElasticVectorSearch\n",
+    "from langchain.vectorstores.opensearch_vector_search import OpenSearchVectorSearch\n",
```
Comment (hwchase17) [inline docs/modules/indexes/chain_examples/vector_db_qa_with_sources.ipynb:24]: probably unnecc

File: docs/modules/indexes/chain_examples/qa_with_sources.ipynb [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "from langchain.vectorstores.elastic_vector_search import ElasticVectorSearch\n",
     "from langchain.vectorstores import Chroma\n",
     "from langchain.docstore.document import Document\n",
-    "from langchain.prompts import PromptTemplate"
+    "from langchain.prompts import PromptTemplate\n",
```
Comment (hwchase17) [inline docs/modules/indexes/chain_examples/qa_with_sources.ipynb:35]: probably unnecc

File: langchain/__init__.py [hunk 3]
```diff
@@ -45,7 +45,7 @@
 from langchain.utilities.google_serper import GoogleSerperAPIWrapper
 from langchain.utilities.searx_search import SearxSearchWrapper
 from langchain.utilities.wolfram_alpha import WolframAlphaAPIWrapper
-from langchain.vectorstores import FAISS, ElasticVectorSearch
+from langchain.vectorstores import FAISS, ElasticVectorSearch, OpenSearchVectorSearch
```
Comment (hwchase17) [inline langchain/__init__.py:48]: lets not do this, trying to encourage putting things at the module level

**Review by hwchase17 — COMMENTED** [review]
some minor nits, but this looks awesome

happy to make the nits as well, we'll get it into tmrws release

File: docs/modules/indexes/chain_examples/vector_db_qa_with_sources.ipynb (hunk 1 above)
Comment (naveentatikonda) [reply docs/modules/indexes/chain_examples/vector_db_qa_with_sources.ipynb]: ack

File: docs/modules/indexes/chain_examples/qa_with_sources.ipynb (hunk 2 above)
Comment (naveentatikonda) [reply docs/modules/indexes/chain_examples/qa_with_sources.ipynb]: ack

File: langchain/__init__.py (hunk 3 above)
Comment (naveentatikonda) [reply langchain/__init__.py]: ack

**Comment by naveentatikonda** [comment]
> some minor nits, but this looks awesome
> 
> happy to make the nits as well, we'll get it into tmrws release

Sounds good. Thanks for reviewing it

**Review by hwchase17 — APPROVED** [review]
awesome :) love this!

**Pull request merged**